tests/rules/method.js

```diff
@@ -450,6 +450,10 @@ eslintTester.run("method", rule, {
             // # 232: disallow setHTMLUnsafe, but OK with static string.
             code: "foo.setHTMLUnsafe('static string')",
         },
+        {
+            code: "(foo as Function)();",
+            ...TYPESCRIPT_OPTIONS,
+        },
```

Comment:
While that is close to my request by the book, I would like to see something that is a bit closer to our linter rule functionality while also allowed.

Maybe `document.write("harmless" as String)` or such?